### DISCUSSION TITLE
Refresh content more easily

### DISCUSSION THREAD

**Discussion by Lakoja** [tuskyapp/Tusky#2563] [Ideas]
Hi and let me first thank you for maintaining and pushing this great app.

I would like to improve the "list update interaction" in certain locations.
But first I'd like to hear if these ideas are going in the right direction (and after that I have some technical difficulties... :-) ).

Currently in order to get to the most recent statuses for example for the home timeline you normally need 4 interactions:
Starting for example from the notifications tab:
* Tap the home timeline tab icon
* Tap it again to get to the top (of the previous old content)
* Swipe down to initiate a refresh: with new status it jumps upwards a bit
* Tap home timeline again to get to the updated top

I'd like this to be two:
* Go to a timeline tab (restores previous position there)
* Tap tab icon again: this goes to the top of the current (new) statuses
This is basically how for example the Twitter app or webpage behaves and also the Mastodon Web UI when "manual refresh" is disabled.

This could be split in two steps:

1. Instantly jump to the top of the list once a swipe-to-refresh is finished (and only then; so not for example if the timeline updates due a new status you just posted)
2. Execute a refresh and jump to new content when the tab icon is tapped again while being at the (old) top

While 1. combines the last two of the above 4 interactions with 2. it would combine the last 3.

What do you think? Is there anything (maybe domain-specific) thing I might have overlooked?

**Comment by Lakoja** [comment]
Technical note: There is an argument to the TimelineFragment ARG_ENABLE_SWIPE_TO_REFRESH that might influence the "jump upwards only a bit". However that is currently not configurable (and probably shouldn't be) and is normally true.

**Comment by charlag** [comment]
Hey, I think it would be good to have fewer interactions however there are a lot of people who want to read from the oldest. It also doesn't work that well at the moment so I think it's worth discussing.

**Reply by Lakoja** [reply]
> a lot of people who want to read from the oldest

So for them it should be no jump to the top? That could probably be a simple configuration option (with of course the question of the default).

But generally - if there is no obvious monetary reason there - I would go with the "mainstream" useability (i. e. Twitter-style).

**Reply by Andrewblasco** [reply]
Personally, I prefer to read from down to top and from oldest to newest. But if it is an option, I will be fine =)
Just my honest and subjective opinion.
